include/chainbase/chainbase.hpp

```diff
@@ -230,6 +230,13 @@ namespace chainbase {
             if( !ok ) BOOST_THROW_EXCEPTION( std::logic_error( "Could not modify object, most likely a uniqueness constraint was violated" ) );
          }
 
+         template<typename Modifier>
+         void replace( const value_type& obj, Modifier&& m ) {
+            on_replace( obj );
```

Comment:
This seems like a pretty dangerous way to implement this. 

`on_replace` will `std::move()` from the existing object, then pass that object into the modifier.  At that point the object is in a "valid by indeterminate state" so, as a caller writing the modifier lambda you have to have very deep and complete knowledge of the object you are replacing.  

The reason I suggested passing a const reference to the old value, and a constructed but un-initialized non-const reference is that it forces the caller to be explicit about the operation rather than relying on implicit safety.  naturally, that would mean that the "std::move" would happen _after_ the Replacer lambda was invoked.

Reply:
I see. I'll fix it.